unstructured_inference/inference/layout.py

```diff
@@ -81,29 +82,43 @@ def from_file(
     ) -> DocumentLayout:
         """Creates a DocumentLayout from a pdf file."""
         logger.info(f"Reading PDF for file: {filename} ...")
-        layouts, images = load_pdf(filename, pdf_image_dpi)
-        if len(layouts) > len(images):
+
+        # Store pdf images for later use
+        output_dir = create_image_output_dir(filename)
+        layouts, _image_paths = load_pdf(
+            filename,
+            pdf_image_dpi,
+            output_folder=output_dir,
+            path_only=True,
+        )
+        image_paths = cast(List[str], _image_paths)
+        if len(layouts) > len(image_paths):
             raise RuntimeError(
-                "Some images were not loaded. Check that poppler is installed and in your $PATH.",
+                "Some images were not loaded. "
+                "Check that poppler is installed and in your $PATH.",
             )
         pages: List[PageLayout] = []
         if fixed_layouts is None:
             fixed_layouts = [None for _ in layouts]
-        for i, (image, layout, fixed_layout) in enumerate(zip(images, layouts, fixed_layouts)):
+        for i, (image_path, layout, fixed_layout) in enumerate(
+            zip(image_paths, layouts, fixed_layouts),
+        ):
             # NOTE(robinson) - In the future, maybe we detect the page number and default
             # to the index if it is not detected
-            page = PageLayout.from_image(
-                image,
-                number=i + 1,
-                detection_model=detection_model,
-                element_extraction_model=element_extraction_model,
-                layout=layout,
-                ocr_strategy=ocr_strategy,
-                ocr_languages=ocr_languages,
-                fixed_layout=fixed_layout,
-                extract_tables=extract_tables,
-            )
-            pages.append(page)
+            with Image.open(image_path) as image:
+                page = PageLayout.from_image(
+                    image,
+                    image_path=image_path,
+                    number=i + 1,
+                    detection_model=detection_model,
+                    element_extraction_model=element_extraction_model,
+                    layout=layout,
+                    ocr_strategy=ocr_strategy,
+                    ocr_languages=ocr_languages,
+                    fixed_layout=fixed_layout,
+                    extract_tables=extract_tables,
+                )
+                pages.append(page)
```

Comment:
>running locally on this branch the process gets killed presumably due to running out of memory

Might `pages` be consuming too much memory here?

Reply:
@six5532one @qued It should not be consuming too much memory here because I set the `image` property of the `PageLayout` instance to `None` to reduce the memory usage - https://github.com/Unstructured-IO/unstructured-inference/blob/fix/pdf2image-too-many-open-files/unstructured_inference/inference/layout.py#L325